internal/levee.go

```diff
@@ -39,75 +41,210 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
-	// TODO: respect configuration scope
 
+	// TODO: respect configuration scope
```

Comment:
Weird whitespace diff.  My moral opposition to `TODO`s says we should just trim this comment if we're touching the area, and open an Issue about being able to limit analysis scope.

Reply:
> we are not respecting configuration scope right now? Do you mean packages? If so I believe we are at least somewhat respecting that right now, is that correct? 🤔

At present, `Config.AnalysisScope` isn't used.  It is reachable through `Config.shouldSkip(pkg)`, but that method isn't called anywhere.

The idea had been that we don't necessarily want to waste cycles scanning every dependency when the sources and sinks we care about are in some higher-level packages.  A call to `conf.shouldSkip(pass.Pkg)` used to be here, but it got replaced by this `TODO` when this project was open-sourced.

By "respect configuration scope," I meant  "we should be able to skip packages that we recognize as irrelevant."

There had been some other hard-coded logic about automatically skipping any package that imports `testing`, since that's presumably a test package.  Looking at it now, there was also a contradicting allow-list that would skip some packages even if they were "in scope."

All this to say, we're due to reexamine both how configuration is provided and how that configuration is consumed to decide which packages to actually analyze.